tests/mapped_struct.py

```diff
@@ -2075,14 +2075,36 @@ class ProxiedDictPackingTest(unittest.TestCase, CollectionPackingTestHelpers, Di
         {'a': 'a2', 'b': 'b2', 'c': 'c2'},
         {1: 10, 2: 20, 3: 30},
         {'a': frozenset(), 'b': (1, 2), 'c': 1.0, 'd': [1, 2], 'e': dict(a=1) },
-        {0: 42, 'a1_@!': 69, 3.5: 'uhhhh', (1, 2, 3): "four-five-six"},
+        {0: 42, 'a1_@!': 69, -3.5: 'uhhhh', (1, 2, 3): "four-five-six"},
         {frozenset([1, 2]) : 97.9},
         {1.0: "test floats equivalent to integers"},
         {None: 3},
         {(1,2,3): 4, (-1,3): 7},
         {frozenset([1,2,3]): 4, frozenset([-1,3]): 7},
+        {float("inf"): 0, float("-inf"): 1, -0.0001: 2, -123456.: 3}
     ]
 
+class StableHashTest(unittest.TestCase):
+    def testHashIntegers(self):
+        values = (1, -1, 1 << 100, -(1 << 100), 1L, -1L)
+        for v in values:
+            self.assertTrue(isinstance(mapped_struct._stable_hash(v), (int, long)))
+
+    def testHashFloats(self):
+        values = (1., -1., 1e+50, 1e-50, float("inf"), float("-inf"), float("nan"))
```

Comment:
What about negative float values with decimal places?

Reply:
`1e-50` has plenty of decimal places, doesn't it ?

Reply:
Yes, but it is a positive value with decimals. I mean a negative value with decimals. Something like ```-1e-50```

Reply:
Got it.